### PR TITLE
[Feature-3034][admin] Support task export

### LINKED ISSUE #3034: [Feature] support task export
### Search before asking

- [X] I had searched in the [issues](https://github.com/DataLinkDC/dinky/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

 support task export

导入:
1. 必须在某目录节点下才有导入按钮
2. 不可直接在没有根目录的情况下导入

导出:
> 统一按照数组格式导出
1. 可导出目录 
4. 可以单独导出某一个任务(以单个数组形式存在)

格式
```json
[
    {
    catalogue :{ catalogue 的全部信息

       },
        task:{
          回归到初识创建作业状态 如下:  -> 其余信息不变根据任务的信息直接导出
            1. 执行模式 local
            5. env 环境 禁用 (-1)
            6. 并行度 1
            7. 全局变量 false
            8. sp 策略
            9. 告警组 禁用((-1)
            10. 版本号 置空
            11. 生命周期-> 开发中


     }
    },
 {
    catalogue :{ catalogue 的全部信息

       },
        task:{
          回归到初识创建作业状态 如下:  -> 其余信息不变根据任务的信息直接导出
            1. 执行模式 local
            5. env 环境 禁用 (-1)
            6. 并行度 1
            7. 全局变量 false
            8. sp 策略
            9. 告警组 禁用((-1)
            10. 版本号 置空
            11. 生命周期-> 开发中


     }
    },
]
```

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by suxinshuo**
[feat-3034][admin] Support task export

This closes #3034

### CODE REVIEW

File: dinky-web/src/pages/DataStudio/LeftContainer/Project/constants.tsx [hunk 1]
```diff
@@ -63,10 +63,14 @@ export const FOLDER_RIGHT_MENU = (disabled = false): MenuItemType[] => [
     label: l('right.menu.rename')
   },
   {
-    key: 'paste',
```
Comment (Zzm0809) [inline dinky-web/src/pages/DataStudio/LeftContainer/Project/constants.tsx]: why remove paste???

**Comment by Zzm0809** [comment]
Can you complete the unit tests?

File: dinky-admin/src/main/java/org/dinky/data/bo/catalogue/export/ExportTaskBO.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    private Integer parallelism;
+
+    private Boolean fragment;
+
+    private Boolean statementSet;
+
+    private Boolean batchModel;
+
+    private Integer clusterId;
+
```
Comment (aiwenmo) [inline dinky-admin/src/main/java/org/dinky/data/bo/catalogue/export/ExportTaskBO.java]: Two platforms will have different IDs, such as clusterId.
How to associate the correct clusterId?

File: dinky-web/src/pages/DataStudio/LeftContainer/Project/constants.tsx (hunk 1 above)
Comment (suxinshuo) [reply dinky-web/src/pages/DataStudio/LeftContainer/Project/constants.tsx]: > why remove paste???

I thought it was an unimplemented import before, now it's restored.

File: dinky-admin/src/main/java/org/dinky/data/bo/catalogue/export/ExportTaskBO.java (hunk 2 above)
Comment (suxinshuo) [reply dinky-admin/src/main/java/org/dinky/data/bo/catalogue/export/ExportTaskBO.java]: image
I extracted a method called resetTask for export tasks and creating a new task. In this method, default values ​​are set for env_id and alert_group_id. I can continue to add other id settings and set them to null. The user can set them manually after importing the task.

Comment (suxinshuo) [reply dinky-admin/src/main/java/org/dinky/data/bo/catalogue/export/ExportTaskBO.java]: But I found a new problem. Is it possible that the two fields firstLevelOwner and secondLevelOwners may correspond to different user ids on different platforms? However, firstLevelOwner is a required field. How should I set the default value for this field? Do you think it is reasonable to set it to the current logged-in user?

Comment (Zzm0809) [reply dinky-admin/src/main/java/org/dinky/data/bo/catalogue/export/ExportTaskBO.java]: 哪个用户导入的就设置为该用户即可, 租户同理

Comment (suxinshuo) [reply dinky-admin/src/main/java/org/dinky/data/bo/catalogue/export/ExportTaskBO.java]: 是的，我也是这个意思，导入操作的时候获取当前操作的登录用户，作为 firstLevelOwner，secondLevelOwners 默认置空

Comment (Zzm0809) [reply dinky-admin/src/main/java/org/dinky/data/bo/catalogue/export/ExportTaskBO.java]: > 是的，我也是这个意思，导入操作的时候获取当前操作的登录用户，作为 firstLevelOwner，secondLevelOwners 默认置空

可

**Comment by suxinshuo** [comment]
> Can you complete the unit tests?

all done. PTAL @Zzm0809 @aiwenmo

**Review by Zzm0809 — APPROVED** [review]

**Review by Zzm0809 — APPROVED** [review]
LGTM

**Pull request merged**